### PR TITLE
Add jtool2

### PR DESCRIPTION
**Pull request by zcutlip**
- jtool2 is a separate tool from jtool, with different capabilities
- Unfortunately the author doesn't publish hashes or version numbers, so
  there's no way to check for versions on this tool

After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] The commit message includes the cask’s name and version.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, if **adding a new cask**:

- [x] Named the cask according to the [token reference].
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.
- [x] Checked there are no [open pull requests] for the same cask.
- [x] Checked the cask was not [already refused].
- [x] Checked the cask is submitted to [the correct repo].

[token reference]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md
[open pull requests]: https://github.com/Homebrew/homebrew-cask/pulls
[already refused]: https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues
[the correct repo]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask
[version-checksum]: https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/sha256.md#updating-the-sha256

### CODE REVIEW

**Comment by vitorgalvao** [comment]
> jtool2 is a separate tool from jtool, with different capabilities

What’s the relationship between them? Why is the only difference in the name a number?

**Comment by zcutlip** [comment]
Levin rewrote jtool from scratch and called it jtool2. He added new capabilities to it, but didn’t port over all the original features. So he distributes both. A little like python’s urllib/urllib2 I guess. So you end up needing both.

More info:
http://newosxbook.com/tools/jtool.html

**Comment by vitorgalvao** [comment]
So `jtool` *is* essentially a new version, which makes this confusing. It means `jtool2` (the tool) should go into `jtool` (the cask), and `jtool` (the tool) should go into [homebrew/cask-versions](https://github.com/Homebrew/homebrew-cask-versions) as `jtool1`. So we’d be left with a `jtool1` cask installing `jtool`, and a `jtool` cask installing `jtool2`.

Not ideal, but it’s the way to keep it consistent. On the other hand, I wonder if this is popular enough to justify having both

```
jtool (added 1103 days ago)
30 days: 43 (#1169)
90 days: 126 (#1135)
365 days: 340 (#1352)
```

Pinging @Homebrew/cask for opinions.

**Comment by zcutlip** [comment]
That could work as long as the two could be installed simultaneously. Levin uses them both in his classes because they do different things. They definitely aren't two versions of the same thing the way, say, a beta or nightly compares to a release version. Or Java 8 vs Java 13.

I'd definitely call the new one 'jtool2' because that's how Levin refers to it in his training. So It would be reasonable to tell someone "install jtool2 with homebrew. You can get jtool there as well if you need it."

I agree; it is confusing and less than ideal. I wish he'd named the two tools differently, or ported all features over to the new tool. Or anything other than "here are two sort of different, sort of the same tools called jtool and jtool2." :-(

Regarding popularity, I don't know how many people even know it's in homebrew-cask.

**Comment by rgov** [comment]
The actual command line tool appears to be called `jtool2` so as a potential user of this cask, I would find it less confusing to install it with `jtool2`. Maybe a caveat is warranted.

**Comment by zcutlip** [comment]
> Edit: I would also be happy with a single `jtool` cask that installs both `jtool` and `jtool2`, until `jtool2` subsumes all features of the original.

This is a good idea and may be the most elegant solution, if homebrew-cask architecture supports it.

**Comment by vitorgalvao** [comment]
> I'd definitely call the new one 'jtool2' because that's how Levin refers to it in his training.

But that’s not how we do it, so either option would confuse someone.

> Regarding popularity, I don't know how many people even know it's in homebrew-cask.

The reason why it isn’t popular is irrelevant. We only care if it’s popular in HBC because it’s here that it’s a maintenance burden for us.

> This is a good idea and may be the most elegant solution, if homebrew-cask architecture supports it.

It doesn’t. If the author packaged both tools in the same archive, it’d be easy.

**Comment by zcutlip** [comment]
At any rate this is a request to add a new program to homebrew-casks. That program is called “jtool2” which has the misfortune of being named similarly to a tool that is different and separate. Anyone familiar with jtool2 will do `brew search jtool2` or `brew cask install jtool2`. `jtool` is a companion tool, albeit similarly named.

If the two tools’ author’s (admittedly confusing) naming scheme prevents their addition to `homebrew-casks` that is disappointing, but not the end of the world.

Further, if the tool’s lack of popularity to date makes it incompatible with `homebrew-casks` that is also disappointing, but not the end of the world.

Please do what is right for your project, and the iOS research community will adapt as necessary.

**Comment by zcutlip** [comment]
At any rate this is a request to add a new program to homebrew-casks. That program is called “jtool2” which has the misfortune of being named similarly to a tool that is different and separate. Anyone familiar with jtool2 will do `brew search jtool2` or `brew cask install jtool2`. `jtool` is a companion tool, albeit similarly named.

If the two tools’ author’s (admittedly confusing) naming scheme prevents their addition to homebrew-casks that is disappointing, but not the end of the world.

Further, if the tool’s lack of popularity to date makes it incompatible with homebrew-casks that is also disappointing, but not the end of the world.

Please do what is right for your project, and the iOS research community will adapt as necessary.

**Comment by vitorgalvao** [comment]
Thank you for your measured and understanding reply, @zcutlip.

The best outcome for both users and maintainers would be if the author had a download on the website that bundled both tools on the same archive (`tgz` or not, doesn’t matter). That way users could install both with a single command and we wouldn’t have a naming issue.

Seeing as the author recommends getting both tools, that might not be a hard sell. Would you be willing to ask?

**Comment by zcutlip** [comment]
> Thank you for your measured and understanding reply, @zcutlip.

No problem. We're all in this together.

> The best outcome for both users and maintainers would be if the author had a download on the website that bundled both tools on the same archive (`tgz` or not, doesn’t matter). That way users could install both with a single command and we wouldn’t have a naming issue.
> 
> Seeing as the author recommends getting both tools, that might not be a hard sell. Would you be willing to ask?

Happy to ask, but past requests have gone unanswered, such as upgrading to https.

**Comment by zcutlip** [comment]
Not getting any movement after requesting the change from the tool’s author. Where does that leave us? Is there no way to have two different but related casks: `jtool` and `jtool2`?

If not, I’d recommend removing `jtool`, and having only a cask called `jtool2`, since ‘2’ isn’t the version number, but part of the tool’s name.

**Comment by zcutlip** [comment]
Funny timing. Got the following response from the author
http://newosxbook.com/forum/viewtopic.php?f=3&t=19739#p24148
Evidently jtool2 v1.0 will be released soon.

**Comment by claui** [comment]
That’s awesome to hear. Thank you @zcutlip for reaching out and following up!

**tl;dr** Jonathan Levin plans to release jtool2 v1.0 next week, which will obviate the need for the old jtool.

I’d suggest that this PR be closed, and then a new PR be filed for the `jtool` cask once the release is out. Would you be willing to file that PR @zcutlip?

**Comment by zcutlip** [comment]
> That’s awesome to hear. Thank you @zcutlip for reaching out and following up!
> 
> **tl;dr** Jonathan Levin plans to release jtool2 v1.0 next week, which will obviate the need for the old jtool.
> 
> I’d suggest that this PR be closed, and then a new PR be filed for the `jtool` cask once the release is out. Would you be willing to file that PR @zcutlip?

This is jtool2 version 1.0, and not jtool version 2.0. I think it needs it's own cask, which is what this PR is for.

Here are a few similar casks:
https://github.com/Homebrew/homebrew-cask/blob/master/Casks/blobby-volley2.rb
https://github.com/Homebrew/homebrew-cask/blob/master/Casks/bitrix24.rb

File: Casks/jtool2.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+cask 'jtool2' do
+  version '1.0.0.0'
+  sha256 :no_check
```
Comment (vitorgalvao) [inline Casks/jtool2.rb]: Will need to be added when the version is released.

**Comment by vitorgalvao** [comment]
> '2' is not a version number, but an intrinsic part of the product name.

That’s fair enough, and it’s the solution I though about when I read [your comment yesterday](https://github.com/Homebrew/homebrew-cask/pull/72669#issuecomment-558753649). Still, the question remains if the author will keep `jtool2` as the name when releasing, considering the other version will be deprecated.

We’ll revisit in a week’s time, when it happens.

**Comment by zcutlip** [comment]
> > '2' is not a version number, but an intrinsic part of the product name.
> 
> That’s fair enough, and it’s the solution I though about when I read [your comment yesterday](https://github.com/Homebrew/homebrew-cask/pull/72669#issuecomment-558753649). Still, the question remains if the author will keep `jtool2` as the name when releasing, considering the other version will be deprecated.
> 
> We’ll revisit in a week’s time, when it happens.

Awesome. Thank you so much for working through this with me. I number of people I know in the iOS research community will benefit from this additional automation when bootstrapping new systems for work and research.

**Comment by zcutlip** [comment]
No release yet. Jonathan often gets slowed down on planned releases for various reasons. Can we move forward with this PR, and revise later if necessary?

**Comment by vitorgalvao** [comment]
Just saw [this tweet](https://twitter.com/Morpheus______/status/1213066557598781440), meaning we can get on with this PR.

**Comment by zcutlip** [comment]
This is great. How do we proceed? Since he still makes jtool available and this tool is called 'jtool2', It seems like we should should go with this separate cask.

**Comment by vitorgalvao** [comment]
> Since he still makes jtool available

It’s available but “officially decommissioned for good”, meaning “don’t use it”. Let’s go with the plan of replacing the `jtool` cask, as that was the point in waiting.

**Comment by rgov** [comment]
So I'll `brew cask install jtool` to get a tool called `jtool2`? Seems unfortunate if that's the way it goes. Just forget about the original `jtool`, and have a formula called `jtool2` that installs a tool called `jtool2`. Is that OK?

**Comment by zcutlip** [comment]
I think the ‘jtool’ cask should be moved to homebrew-cask-versions, and keep the jtool2 cask here

**Comment by vitorgalvao** [comment]
It’s getting tiring to discuss this issue. Quite frankly, considering the (lack of) popularity of this cask (it’s not a formula), it’s way past the threshold where it’s worth the time investment.

While you’re quite happy to only have to think of yourselves, maintainers have to think about project consistency as a whole.

Yes, it sucks that on occasion it leads to slightly weird decisions (and ”slightly” is the operative word here), but we fix those if they reveal themselves to be problematic, *which they seldom do*. You two have the biggest incentive to not care about the cask’s token—you know what it is and what it represents.

I’ve worked with you way more than this cask warrants to reach an acceptable solution where it could be included in the main repository and keep consistency instead of being outright refused. We’ve finally reached that stage and you know what needs to be done to have it included. After you’re long gone, we’ll be the ones left having to do maintenance; I’m not going to break project conventions and make a confusing exception for an unpopular cask.

If you’re that unhappy with the name decision, you’re welcome to host your own tap. The fact that no one is beholden to Homebrew for what they can install is one of my favourite features.

**Comment by zcutlip** [comment]
My apologies. I didn’t intend for this to go on as long as it has. I was just suggesting, as a way to proceed, we go with what we settled on here:
https://github.com/Homebrew/homebrew-cask/pull/72669#issuecomment-559197384

That seemed easiest and there was consensus, but please do whatever’s right for your project.

**Comment by vitorgalvao** [comment]
You’re right, @zcutlip. I was harsher than I should because despite having reread that post (I had forgotten the conversation), I’ve misunderstood the stage we were in.

For that reason, I’ve retracted my earlier post. Please accept my apologies.

Could you please add the missing `sha256`?

**Comment by zcutlip** [comment]
> You’re right, @zcutlip. I was harsher than I should because despite having reread that post (I had forgotten the conversation), I’ve misunderstood the stage we were in.
> 
> For that reason, I’ve retracted my earlier post. Please accept my apologies.

No worries. You've been very patient.

> Could you please add the missing `sha256`?

I updated the cask to use `version :latest`. Sadly, he doesn't version the tarball. I've asked if he'll consider it, but I'm not going to hold my breath. He doesn't do that for any of his tools.

**Comment by vitorgalvao** [comment]
> Sadly, he doesn't version the tarball.

How often is the software updated? If it’s sporadic (let’s say once a month or less) there’s no problem in adding a `version` and `sha256` anyway. When the latter fails, we can fix the cask.

**Comment by vitorgalvao** [comment]
> Sadly, he doesn't version the tarball.

How often is the software updated? If it’s sporadic (didn’t the author say somewhere this would be the first and last version? Or am I thinking of something else?) there’s no problem in adding a `version` and `sha256` anyway. When the latter fails, we can fix the cask.

**Comment by zcutlip** [comment]
Yes I'd say it's sporadic. Sometimes a week or so in between, but I'd say on average about 1-2 months.

I'll add the version and checksum back in. Since he doesn't really use version numbers, I'll use a number based on the release date, like `2020.01.01`. He seems pretty consistent about including each release date in WhatsNew.txt.

I'm happy to be on the hook for updating the cask. Is there a way to be notified if it breaks?

**Comment by vitorgalvao** [comment]
> Yes I'd say it's sporadic. Sometimes a week or so in between, but I'd say on average about 1-2 months.

It’s fine, then. Let’s try it out and if it becomes problematic we can switch to `:latest`/`:no_check` at that time.

> Since he doesn't really use version numbers, I'll use a number based on the release date, like `2020.01.01`. He seems pretty consistent about including each release date in WhatsNew.txt.

Sounds good.

> I'm happy to be on the hook for updating the cask.

Thank you.

> Is there a way to be notified if it breaks?

If you `brew cask fetch` it, the command will return `0` if the download is successful (including correct checksum) and `1` otherwise. But it might be more efficient to monitor releases upstream (e.g. if `WhatsNew.txt` is published online).

**Comment by zcutlip** [comment]
Okay, done. Let me know what else is required.

**Review by vitorgalvao — APPROVED** [review]

**Comment by zcutlip** [comment]
Thank you for merging this and for working with me to get it done.

Ugh, it looks like he's doing nightly builds. Maybe automated? Because the hash of the tarball and of the binary keeps changing, but the `WhatsNew.txt` isn't changing. I'll keep an eye on it, and possibly submit a PR removing the sha256 until the builds stop.
